utils/fs/directory.cpp

```diff
@@ -127,15 +127,9 @@ struct utils::fs::detail::directory_iterator::impl : utils::noncopyable {
     /// not.  A null pointer means an invalid iterator.
     ::DIR* _dirp;
 
-    /// Raw representation of the system directory entry.
-    ///
-    /// We need to keep this at the class level so that we can use the
-    /// readdir_r(3) function.
-    ::dirent _dirent;
-
     /// Custom representation of the directory entry.
     ///
-    /// This is separate from _dirent because this is the type we return to the
+    /// This is separate from readdir because this is the type we return to the
```

Comment:
`_dirent` used to refer to a specific field. "separate from readdir" now doesn't make a lot of sense. Please reword.